src/config.rs

```diff
@@ -1742,6 +1743,23 @@ mod tests {
         // Sanity check that key bindings are being parsed
         assert!(!config.key_bindings.is_empty());
     }
+
+    #[test]
+    fn update_dynamic_title() {
+        let config: Config = ::serde_yaml::from_str(ALACRITTY_YML)
+            .expect("deserialize config");
+        assert!(config.dynamic_title);
+
+        let mut options = Options::default();
+        let config = config.update_dynamic_title(&options);
+        assert!(config.dynamic_title);
+        options.title = Some("".to_owned());
+        let config = config.update_dynamic_title(&options);
+        assert!(!config.dynamic_title);
+        options.title = Some("foo".to_owned());
+        let config = config.update_dynamic_title(&options);
+        assert!(!config.dynamic_title);
```

Comment:
These two tests don't really test anything different, we there's no logic anywhere that checks the string in any way, so an empty string isn't really more of a special case than a full one, I'd probably just go with `"foo"` and assert that just that will read to dynamic title being disabled.